### DISCUSSION TITLE
propagate  custom baggages to downstream services.

### DISCUSSION THREAD

**Discussion by abhiviv** [signalfx/splunk-otel-java#1523] [Q&A]
Our app is spring boot based. Earlier we were using Spring cloud slueth for trace and baggage propogation.
Recently we started used Splunk OTEL instrumentation agent (java) and removed slueth library. Traces are getting propagated properly.
but could not find any option to propagate  custom baggages to downstream services.

Example, Caller of my app passes "X-REQUEST-ID" in headers in API call. I want to pass the same header to other downstream services my app is calling. I believe X-REQUEST-ID is considered as custom baggage so looking for your help on how to configure that in splunk otel agent library.

**Comment by hshah2811** [comment]
Facing same issue. can anyone provide solution?

**Comment by laurit** [comment]
Sorry for the late reply. I believe you can achieve this with a custom propagator, for inspiration see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/extension/src/main/java/com/example/javaagent/DemoPropagator.java
